tests/drivers/led/led_api/src/test_led_api.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <drivers/led.h>
+
+#define BRIGHTNESS_MAX	100
+#define TEST_MAX_COLORS	8
+#define COLOR_FULL	0xff
+
+#if DT_NODE_HAS_STATUS(DT_ALIAS(led_controller_0), okay)
+#define LED_CTRL_NODE_ID	DT_ALIAS(led_controller_0)
+#define LED_CTRL_DEV_NAME	DT_LABEL(LED_CTRL_NODE_ID)
+#else
+#error "LED controller device not found"
+#endif
+
+#define _COLOR_MAPPING(led_node_id)				\
+const uint8_t test_color_mapping_##led_node_id[] =		\
+	DT_PROP(led_node_id, color_mapping)
+
+#define COLOR_MAPPING(led_node_id)					\
+	IF_ENABLED(DT_NODE_HAS_PROP(led_node_id, color_mapping),	\
+		   (_COLOR_MAPPING(led_node_id);))
+
+#define LED_INFO_COLOR(led_node_id)				\
+{								\
+	.label		= DT_LABEL(led_node_id),		\
+	.index		= DT_PROP_OR(led_node_id, index, 0),	\
+	.num_colors	=					\
+		DT_PROP_LEN(led_node_id, color_mapping),	\
+	.color_mapping	= test_color_mapping_##led_node_id,	\
+},
+
+#define LED_INFO_NO_COLOR(led_node_id)				\
+{								\
+	.label		= DT_LABEL(led_node_id),		\
+	.index		= DT_PROP_OR(led_node_id, index, 0),	\
+	.num_colors	= 0,					\
+	.color_mapping	= NULL,					\
+},
+
+#define LED_INFO(led_node_id)	\
+	COND_CODE_1(DT_NODE_HAS_PROP(led_node_id, color_mapping),	\
+		    (LED_INFO_COLOR(led_node_id)),			\
+		    (LED_INFO_NO_COLOR(led_node_id)))
+
+#define LED_CONTROLLER_INFO(node_id)				\
+								\
+DT_FOREACH_CHILD(node_id, COLOR_MAPPING)			\
+								\
+const struct led_info test_led_info[] = {			\
+	DT_FOREACH_CHILD(node_id, LED_INFO)			\
+};								\
+								\
+static int num_leds = ARRAY_SIZE(test_led_info)
```

Comment:
user mode will not be able to access this memory without being routed to a memory domain partition.

by far the easiest way to do this is to prefix all non-const globals with ZTEST_DMEM or ZTEST_BMEM appropriately for data and bss.

This particular one would be ZTEST_DMEM since it's initialized data.

No need to do this for const data as it's globally accessible.